logot/_structlog.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from functools import partial
+from typing import Any, MutableMapping
+
+import structlog
+from structlog.exceptions import DropEvent
+
+from logot._capture import Captured
+from logot._logot import Capturer, Logot
+from logot._typing import Level, Name
+
+EventDict = MutableMapping[str, Any]
+WrappedLogger = Any
+
+
+CRITICAL = 50
+FATAL = CRITICAL
+ERROR = 40
+WARNING = 30
+WARN = WARNING
+INFO = 20
+DEBUG = 10
+NOTSET = 0
+
+NAME_TO_LEVEL = {
+    "critical": CRITICAL,
+    "exception": ERROR,
+    "error": ERROR,
+    "warn": WARNING,
+    "warning": WARNING,
+    "info": INFO,
+    "debug": DEBUG,
+    "notset": NOTSET,
+}
+
+
+class StructlogCapturer(Capturer):
+    """
+    A :class:`logot.Capturer` implementation for :mod:`structlog`.
+    """
+
+    __slots__ = ("_old_processors", "_old_wrapper_class")
+
+    def start_capturing(self, logot: Logot, /, *, level: Level, name: Name) -> None:
+        config = structlog.get_config()
+        processors = config["processors"]
+        wrapper_class = config["wrapper_class"]
+        self._old_processors = processors.copy()
+        self._old_wrapper_class = wrapper_class
+        processors.clear()
+        processors.append(partial(_processor, logot=logot, name=name))
+
+        if level is not None:
+            if isinstance(level, str):
+                levelno = NAME_TO_LEVEL[level.lower()]
+                wrapper_class = structlog.make_filtering_bound_logger(levelno)
```

Comment:
```suggestion
        if isinstance(level, str):
            level = NAME_TO_LEVEL[level.lower()]
        wrapper_class = structlog.make_filtering_bound_logger(level)
```
`level` is never `None`, and we always want to make the filtering bound logger for both `str` or `int` levels I think?

Reply:
I don't know structlog, but is there a reason to not implement the filtering in the processor?

Reply:
Recommended approach specifically for log level is this https://www.structlog.org/en/stable/bound-loggers.html#filtering-by-log-levels

Reply:
Interesting. That's a performance argument. But that seems to set the level for *all* processors, and ideally we'd leave the rest of the processor chain untouched. This is particularly important if multiple captures are used.

Reply:
Have switched this to filtering in the `_processor` 👍